src/CakeSlayer/Password.hs

```diff
@@ -0,0 +1,74 @@
+-- | User passwords. Uses @bcrypt@ password hashing.
+
+module CakeSlayer.Password
+       ( PasswordHash (unPasswordHash)
+       , PasswordPlainText (..)
+       , unsafePwdHash
+       , mkPasswordHashWithPolicy
+       , mkPasswordHash
+       , verifyPassword
+       , generateRandomPassword
+       ) where
+
+import Data.Aeson (FromJSON, ToJSON)
+import Database.PostgreSQL.Simple.FromField (FromField)
+import Database.PostgreSQL.Simple.ToField (ToField)
+import Elm (Elm)
+
+import CakeSlayer.Error (WithError, throwOnNothingM)
+import CakeSlayer.Random (mkRandomString)
+
+import qualified Crypto.BCrypt as BC
+
+
+-- | Password hash.
+newtype PasswordHash = PasswordHash
+    { unPasswordHash :: Text
+    } deriving stock (Show, Generic)
+      deriving newtype (Eq, FromField, ToField, FromJSON, ToJSON, Elm)
```

Comment:
Is there any value in not deriving the `FromJSON` and `ToJSON` instances here? Password hash's should probably never be sent to a frontend (or atleast I can't think of many cases where this might be needed).

If we leave out this derivation, it will stop a user from incorrectly serializing something like a `User` type which stores a `PasswordHash`.

Reply:
I don't mind removing `FromJSON`, `ToJSON` and `Elm` instances from `PasswordHash`. Probably it's indeed should never been serialized. I just tried to remove those instances from backend and looks like they are needed only for `User` and `User` ones are not used.